src/effects/nyquist/Nyquist.cpp

```diff
@@ -117,6 +117,68 @@ enum
 static const wxChar *KEY_Command = wxT("Command");
 static const wxChar *KEY_Parameters = wxT("Parameters");
 
+///////////////////////////////////////////////////////////////////////////////
+//
+// NyquistEffectUIValidator
+//
+///////////////////////////////////////////////////////////////////////////////
+struct NyquistEffect::SharedValidatorState final
+{
+   explicit SharedValidatorState(NyquistEffect& effect)
+       : mEffect(&effect), mParent(mEffect->GetUIParent())
+   {
+      mParent->PushEventHandler(&effect);
+   }
+
+   ~SharedValidatorState()
+   {
+      Reset();
+   }
+
+   void Reset()
+   {
+      if (mEffect == nullptr)
+         return;
+
+      if (mParent)
+         mParent->PopEventHandler();
+
+      mEffect = nullptr;
+   }
+
+   NyquistEffect* mEffect;
+   wxWeakRef<wxWindow> mParent;
```

Comment:
Make the members const?  Or hold a reference not pointer to effect?

Reply:
I specifically use pointer so I can nullify it on Reset.